File: app/src/main/java/com/sonu/app/splash/ui/photo/PhotoListItem.java
package com.sonu.app.splash.ui.photo;

import com.sonu.app.splash.model.unsplash.Photo;
import com.sonu.app.splash.ui.list.ListItem;
import com.sonu.app.splash.ui.list.ListItemTypeFactory;

/**
 * Created by amanshuraikwar on 20/12/17.
 */

public class PhotoListItem extends ListItem<PhotoOnClickListener> {

    private Photo photo;

    public PhotoListItem(Photo photo) {
        this.photo = photo;
    }

    public Photo getPhoto() {
        return photo;
    }

    @Override
    public int type(ListItemTypeFactory typeFactory) {
        return PhotoListItem.class.hashCode();
    }
}

File: app/src/main/java/io/github/amanshuraikwar/splash/ui/list/photo/PhotoListItem.kt
package io.github.amanshuraikwar.splash.ui.list.photo

import io.github.amanshuraikwar.multiitemlistadapter.ListItem
import io.github.amanshuraikwar.multiitemlistadapter.SimpleListItemOnClickListener
import io.github.amanshuraikwar.splash.model.Photo
import io.github.amanshuraikwar.splash.ui.list.ListItemTypeFactory

/**
 * ListItem representing a CryptoCurrency in the RecyclerView.
 *
 * @author Amanshu Raikwar
 * Created by amanshuraikwar on 11/03/18.
 */
class PhotoListItem(val photo: Photo)
    : ListItem<SimpleListItemOnClickListener, ListItemTypeFactory>() {

    override fun id() = photo.id!!

    override fun concreteClass() = PhotoListItem::class.toString()

    override fun type(typeFactory: ListItemTypeFactory): Int {
        return typeFactory.type(this)
    }
}